subsys/logging/Kconfig.template.log_format_config

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2022 Intel Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config LOG_BACKEND_$(backend)_OUTPUT_DICTIONARY
+	bool
+	depends on LOG2
+	select LOG_DICTIONARY_SUPPORT
+	help
+	  Backend is in dictionary-based logging output mode. Supported only for UART backend.
```

Comment:
"Supported only for UART backend?" still true?

Reply:
@nordic-krch Prior this implementation, this flag only existed for UART backend. Also the support for Dictionary Format as well was only in UART backend. This Kconfig template would enable all other backends as well for dictionary format, although I wasn't sure if the support can be validated therefore provided this helper info just incase. Thoughts?